src/core/annotation.js

```diff
@@ -237,32 +303,6 @@ var Annotation = (function AnnotationClosure() {
       }
     },
 
-    isInvisible: function Annotation_isInvisible() {
-      var data = this.data;
-      return !!(data &&
-                data.annotationFlags &&            // Default: not invisible
-                data.annotationFlags & 0x1);       // Invisible
-    },
-
-    isViewable: function Annotation_isViewable() {
-      var data = this.data;
-      return !!(!this.isInvisible() &&
-                data &&
-                (!data.annotationFlags ||
-                 !(data.annotationFlags & 0x22)) &&  // Hidden or NoView
-                data.rect);                          // rectangle is necessary
```

Comment:
We do not need to check for `data.rect` anymore. Previously this was necessary because `data.rect` could be `undefined` or `null`, but after recent refactoring patches this cannot happen anymore. When an annotation is constructed, https://github.com/mozilla/pdf.js/blob/master/src/core/annotation.js#L126 is always called, causing https://github.com/mozilla/pdf.js/blob/master/src/core/annotation.js#L152 to happen when a rectangle is invalid. This means that there is always a valid rectangle array: either the one from the PDF or `[0,0,0,0]`.